### DISCUSSION TITLE
PyTorch nn.Conv2D to Flax linen.Conv

### DISCUSSION THREAD

**Discussion by kamalkraj** [google/flax#1680] [Q&A] [answered]
Hi,

I am trying to move a model from PyTorch to Flax. 

Is this correct translation for the 2D convolution layer from PyTorch to Flax

PyTorch
```python
import torch.nn as nn
conv = nn.Conv2d(1, 32, kernel_size=(41, 11), stride=(2, 2), padding=(20, 5))

```

Flax
```python
from flax import linen as nn
conv = nn.Conv(features=32, kernel_size=(41, 11), strides=(2, 2), padding=((0, 20), (0, 5)))
```

I am confused about the padding parameter?

**Comment by marcvanzee** [comment, marked as answer]
Have your checked the documentation? I think it is actually quite clear:

- From the [Pytorch documentation](https://pytorch.org/docs/stable/generated/torch.nn.Conv2d.html): "`padding` controls the amount of padding applied to the input. It can be either a string {‘valid’, ‘same’} or a tuple of ints giving the amount of implicit padding applied on both sides."
- From the [Flax documentation](https://flax.readthedocs.io/en/latest/_autosummary/flax.linen.Conv.html#flax.linen.Conv.padding): "either the string ‘SAME’, the string ‘VALID’, the string ‘CIRCULAR’` (periodic boundary conditions), or a sequence of n (low, high) integer pairs that give the padding to apply before and after each spatial dimension."

So the `torch` Conv2d you provided above can be translated to Flax as follows:

```python
 nn.Conv(features=32, kernel_size=(41, 11), strides=(2, 2), padding=((20, 20), (5, 5)))
```

**Reply by andsteing** [reply]
If you're porting PyTorch code you might want to check out this thread:
https://github.com/google/flax/discussions/1770#discussioncomment-1940313
